s3torchconnector/tst/unit/test_s3dataset_common.py

```diff
@@ -69,16 +69,13 @@ def test_list_objects_from_prefix(
     prefix: str, keys: Sequence[str], expected_count: int
 ):
     mock_client = _create_mock_client_with_dummy_objects(TEST_BUCKET, keys)
-    objects = list_objects_from_prefix(f"{S3_PREFIX}/{prefix}", mock_client)
+    bucket_key_pairs = list_objects_from_prefix(f"{S3_PREFIX}/{prefix}", mock_client)
     count = 0
-    for index, object in enumerate(objects):
+    for index, bucket_key_pair in enumerate(bucket_key_pairs):
         count += 1
```

Comment:
Why do we still have count? Shouldn't index be enough?

Reply:
I think, we need `count` to prevent situation when number of returned objects will be smaller than number of `expected_keys`. We use `count` in assert below the "for" loop.